### PR TITLE
feat: Fable model usage tracking — scoped threshold notifications (Story 21.4)

### PR DESCRIPTION
**Pull request by rajish**
## Summary

Story 21.4 of Epic 21 (Fable Model Usage Tracking): threshold notifications for the model-scoped weekly cap.

- Third threshold state machine in `NotificationService`, beside the existing 5h/7d ones, fed the first `weekly_scoped` limit each poll cycle and on preference-change re-evaluation
- Silent warning below the configured warning threshold, sounded critical below the critical threshold (defaults 20%/5%), once per crossing, re-arming on recovery, fully independent of the other windows
- Notification body labeled with the API display name ("Fable", generic "Model" fallback), carrying countdown plus absolute reset time; stable `headroom-warning-scoped` / `headroom-critical-scoped` identifiers
- Accounts without a scoped limit see zero behavior change; existing 5h/7d notification bodies, identifiers, and semantics are byte-for-byte unchanged

## Changes

- `NotificationServiceProtocol` — `scoped: ScopedLimitState?` parameter on `evaluateThresholds`, `scopedThresholdState` accessor
- `NotificationService` — scoped state machine reusing `evaluateWindow` unchanged; scoped clause in the threshold-change re-arm block; delivery refactored to caller-supplied body prefix (5h/7d output unchanged)
- `PollingEngine` — forwards the first scoped entry (nil when absent)
- Tests: 21 scoped state-machine tests covering the full edge-case matrix, poll-cycle wiring test, mock conformance; two pre-existing vacuous absolute-time assertions tightened

## Review

Adversarial code review (4 layers) completed: 3 low-severity patches applied, 1 pre-existing issue deferred (sign-out does not reset threshold dedup state — logged in deferred-work.md), 13 findings dismissed as noise. Details in the story spec's Review Findings and Review Triage Log.

## Testing

- `xcodebuild test`: 1542 tests in 129 suites, 0 failures (20 tests added by the story, 2 by review patches)
- Every I/O edge-case matrix row in the spec is covered by a test that ran and passed

## Summary by CodeRabbit

* **New Features**
  * Added threshold notifications for Fable’s model-specific weekly limit.
  * Notifications support warning and critical levels, recovery, and re-arming after limits change.
  * Fable limit information is refreshed during polling and displayed with readable labels and reset times.
  * Added fallback handling when limit names or reset dates are unavailable.

* **Documentation**
  * Completed documentation and acceptance criteria for Fable limit notifications and tracking.

### CODE REVIEW

File: _bmad-output/implementation-artifacts/spec-21-4-fable-threshold-notifications.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+- Given the user changes warning/critical thresholds in Settings, when `reevaluateThresholds` runs, then the scoped machine re-arms/fires against the new values exactly like the 5h/7d machines.
+
+### Review Findings
+
+- [x] [Review][Patch] I/O matrix row "Threshold prefs changed" states a self-contradictory illustrative outcome; corrected to the implemented 5h/7d-parity behavior [_bmad-output/implementation-artifacts/spec-21-4-fable-threshold-notifications.md]
+- [x] [Review][Patch] Auto Run Result "Files changed" list omitted the regenerated `epic-21-context.md`; added [_bmad-output/implementation-artifacts/spec-21-4-fable-threshold-notifications.md]
+- [x] [Review][Patch] Two pre-existing vacuous absolute-time assertions `contains("at ")` in the 5h/7d body tests tightened to `contains("(at ")` [cc-hdrmTests/Services/ThresholdStateMachineTests.swift:247]
+- [x] [Review][Defer] Threshold dedup state (5h, 7d, and now scoped) survives sign-out; a different account signing in inherits the previous account's warned state [cc-hdrm/App/AppDelegate.swift:373] — deferred, pre-existing
```
[the remark this answers is not in the document]
Comment (rajish) [reply _bmad-output/implementation-artifacts/spec-21-4-fable-threshold-notifications.md:110]: Acknowledged — keeping the `[Review][Patch]` / `[Review][Defer]` bracket prefixes as-is: this is the exact findings format prescribed by the BMAD code-review workflow template, and this file is an internal implementation artifact rather than rendered documentation. Resolving.
